### PR TITLE
Mention native platform update in upgrade notes

### PR DESCRIPTION
**Pull request by wolfs**
Fixes https://github.com/gradle/gradle-private/issues/2981.

### CODE REVIEW

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_6.adoc [hunk 1]
```diff
@@ -57,6 +57,14 @@ The dependencies for annotation processors are not added to an IDEA module's cla
 - PMD has been updated to https://pmd.github.io/pmd-6.21.0/pmd_release_notes.html#24-january-2020---6210[PMD 6.21.0].
 - CodeNarc has been updated to https://github.com/CodeNarc/CodeNarc/blob/v1.5/CHANGELOG.md#version-15----nov-2019[CodeNarc 1.5].
 
+==== Degraded experience when using Gradle on 32-bit operating systems
```
Comment (lptr) [inline subprojects/docs/src/docs/userguide/migration/upgrading_version_6.adoc]: I think this is much too strong a title. We also don't drop support for all 32-bit OSs.

I'd focus on the visible changes, and go with a softer approach, like:

```suggestion
==== Rich console support discontinued for some 32-bit operating systems
```

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_6.adoc [hunk 2]
```diff
@@ -57,6 +57,14 @@ The dependencies for annotation processors are not added to an IDEA module's cla
...
+
+Gradle uses the https://github.com/gradle/native-platform[`native-platform`] library to call native platform APIs for things which are not easily possible from plain Java.
+Most notably, it uses `native-platform` for a better console experience.
```
Comment (lptr) [inline subprojects/docs/src/docs/userguide/migration/upgrading_version_6.adoc]: I think this is too much technical detail here. I think we should drop this paragraph.

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_6.adoc [hunk 3]
```diff
@@ -57,6 +57,14 @@ The dependencies for annotation processors are not added to an IDEA module's cla
...
+
+Gradle 6.3 uses `native-platform` 0.22, which drops support for 32-bit Unix systems and for old FreeBSD versions (older than FreeBSD 10).
+While Gradle will still continue building your projects on that systems, you may have a worse user experience especially when it comes to console output.
```
Comment (lptr) [inline subprojects/docs/src/docs/userguide/migration/upgrading_version_6.adoc]: Again, too much technical detail. The cause of the change doesn't make a difference for the user, as they can't do anything about it. Why not just write:

```suggestion
Gradle 6.3 does not support the rich console for 32-bit Unix systems and for old FreeBSD versions (older than FreeBSD 10).
Gradle will continue building your projects on the affected systems, but without the rich console.
32-bit versions of Windows are unaffected, and will keep working as before.
```

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_6.adoc (hunk 2 above)
Comment (wolfs) [reply subprojects/docs/src/docs/userguide/migration/upgrading_version_6.adoc]: I dropped the paragraph.

**Comment by wolfs** [comment]
@lptr PTAL!

I am wondering if the rich console
- works without native platform
- is the only thing that will stop working on 32bit systems.

Do you have any ideas about ☝️ ?

**Comment by lptr** [comment]
My understanding is that the rich console should have the only visible changes. The rest should be replacements for Java 7 I/O APIs, but those should only be reflected in performance, and it's unclear if they will be improvements or regressions.

That said, my understanding of `native-platform` is limited. Maybe @adammurdoch could clarify?

**Pull request merged**